libraries/chain/include/eosio/chain/name.hpp

```diff
@@ -18,24 +20,36 @@ namespace eosio::chain {
          return (c - 'a') + 6;
       if( c >= '1' && c <= '5' )
          return (c - '1') + 1;
+      else if( c == '.')
+         return 0;
+      else
```

Comment:
no need to else here, directly call assert, and remove the unreachable return value